rmgpy/data/kinetics/family.py

```diff
@@ -1602,6 +1604,24 @@ def apply_recipe(self, reactant_structures, forward=True, unique=True):
                 lowest_labels.append(min(labels))
             product_structures = [s for _, s in sorted(zip(lowest_labels, product_structures))]
 
+        # if applying the family in reverse and the template reactants restrict multiplicity, we need to make sure that
+        # a reverse product matches the multiplicity-constrained template reactant because the forward template products
+        # do not enforce the multiplicity restriction
+        if not forward and reactant_type == 'mol':
+            multiplicity_restricted_templates = []
+            for node in self.forward_template.reactants:
+                if isinstance(node.item, Group):
+                    if node.item.multiplicity != []:
+                        multiplicity_restricted_templates.append(node)
+            if multiplicity_restricted_templates:
```

Comment:
I think this `if` is redundant. If the list is empty the following `for` will just run zero times.